elastic/apm-server
# Add {span,transaction}.representative_count fields

## Motivation/summary

If RepresentativeCount is set (>0), store it as `{span,transaction}.representative_count` in the span/transaction documents. This may be used in queries for calculating the effective sample rate over a time range, or for calculating weighted averages from trace events.

## Checklist

~- [ ] Documentation has been updated~

## How to test these changes

1. Configure an agent with sample rate 0.016
2. Trigger requests that will cause transactions and spans to be sent to APM Server
3. Check that each transaction and span document has a `transaction.representative_count` or `span.representative_count` field with value 62.5.

## Related issues

Closes https://github.com/elastic/apm-server/issues/9340
